lib/fluent/plugin/out_http.rb

```diff
@@ -236,11 +269,28 @@ def create_request(chunk, uri)
             when :put
               Net::HTTP::Put.new(uri.request_uri)
             end
+      set_headers(req, uri, chunk)
+      req.body = @json_array ? "[#{chunk.read.chop}]" : chunk.read
+
       if @auth
-        req.basic_auth(@auth.username, @auth.password)
+        if @auth.method == :basic
+          req.basic_auth(@auth.username, @auth.password)
+        elsif @auth.method == :aws_sigv4
+          signature = @aws_signer.sign_request(
+            http_method: req.method,
+            url: uri.request_uri,
+            headers: {
+              'Content-Type' => @content_type,
+              'Host' => uri.host
+            },
+            body: req.body
+          )
+          req.add_field('x-amz-date', signature.headers['x-amz-date'])
+          req.add_field('x-amz-security-token', signature.headers['x-amz-security-token'])
+          req.add_field('x-amz-content-sha256', signature.headers['x-amz-content-sha256'])
+          req.add_field('authorization', signature.headers['authorization'])
+        end
       end
```

Comment:
Since there are more code lines for `auth`, it would be better to extract them into a new method, such as `set_auth`. (like `set_headers`)

Reply:
Resolved.